packages/@aws-cdk/core/lib/cfn-output.ts

```diff
@@ -51,6 +51,10 @@ export class CfnOutput extends CfnElement {
 
     if (props.value === undefined) {
       throw new Error(`Missing value for CloudFormation output at path "${this.node.path}"`);
+    } else if (Array.isArray(props.value)) {
+      // `props.value` is a string, but because cross-stack exports allow passing any,
+      // we need to check for lists here.
+      throw new Error(`CloudFormation output was given a string list instead of a string at path "${this.node.path}"`);
```

Comment:
Does this definitely fail in CloudFormation, or does it implicitly join or something?

Reply:
it definitely fails